climada/engine/unsequa/calc_delta_climate.py

```diff
@@ -0,0 +1,435 @@
+    """
+
+    _input_var_names = (
+        "exp_initial_input_var",
+        "impf_initial_input_var",
+        "haz_initial_input_var",
+        "exp_final_input_var",
+        "impf_final_input_var",
+        "haz_final_input_var",
+    )
+    """Names of the required uncertainty variables"""
+
+    _metric_names = ("aai_agg", "freq_curve", "at_event", "eai_exp")
+    """Names of the cost benefit output metrics"""
+
+    def __init__(
+        self,
+        exp_initial_input_var: Union[InputVar, Exposures],
+        impf_initial_input_var: Union[InputVar, ImpactFuncSet],
+        haz_initial_input_var: Union[InputVar, Hazard],
+        exp_final_input_var: Union[InputVar, Exposures],
+        impf_final_input_var: Union[InputVar, ImpactFuncSet],
+        haz_final_input_var: Union[InputVar, Hazard],
+    ):
+        """Initialize UncCalcImpact
+
+        Sets the uncertainty input variables, the impact metric_names, and the
+        units.
+
+        Parameters
+        ----------
+        exp_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure of initial state
+        impf_initital_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set of initial state
+        haz_initial_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard of initial state
+        exp_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure of final state
+        impf_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set of final state
+        haz_final_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard of final state
+
+        """
+
+        Calc.__init__(self)
+        self.exp_initial_input_var = InputVar.var_to_inputvar(exp_initial_input_var)
+        self.impf_initial_input_var = InputVar.var_to_inputvar(impf_initial_input_var)
+        self.haz_initial_input_var = InputVar.var_to_inputvar(haz_initial_input_var)
+        self.exp_final_input_var = InputVar.var_to_inputvar(exp_final_input_var)
+        self.impf_final_input_var = InputVar.var_to_inputvar(impf_final_input_var)
+        self.haz_final_input_var = InputVar.var_to_inputvar(haz_final_input_var)
+
+        self.value_unit = self.exp_initial_input_var.evaluate().value_unit
+        self.check_distr()
+
+    def uncertainty(
+        self,
+        unc_sample,
+        rp=None,
+        calc_eai_exp=False,
+        calc_at_event=False,
+        processes=1,
+        chunksize=None,
```

Comment:
If you have the time, could you add type hints?

Reply:
here too, I mirrored `CalcImpact`. I give the ball back to you and @chahank for such modifications.

Reply:
No worries. Let's leave it like that.